src/lib/updating-element.ts

```diff
@@ -304,7 +312,7 @@ export abstract class UpdatingElement extends HTMLElement {
       return;
     }
     const key = typeof name === 'symbol' ? Symbol() : `__${name}`;
-    Object.defineProperty(this.prototype, name, {
+    let descriptor: PropertyDescriptor|null|void = {
```

Comment:
Is `descriptorFactory` expected to mutate the passed in descriptor, or return a new one? Rather than passing a description, what's the information needed to create a new one?

Reply:
Either. To create an entirely new descriptor, one mainly needs the `PropertyDeclaration` provided in `options`. The `key` is available so that the user can easily follow LitElement's convention for private property storage.

The previous descriptor is provided because in some of the use cases evaluated (e.g. making a property validator), only the setter needs to be customized.

Reply:
Removed factory in favor of `createPropertyDescriptor`.